universal/iservices.md

```diff
@@ -1,10 +1,119 @@
+
+The following items will be created below and are required to use iServices:
+
+- MLB
+- ROM*
+- SystemProductName
+- SystemSerialNumber
+- SystemUUID
+
+\**for ROM, we use the MAC Address of the network interface, lowercase, and without `:`.*
 
 **Note**: You and you alone are responsible for your AppleID, read the guide carefully and take full responsibility if you screw up. Dortania and other guides are not held accountable for what **you** do.
 
-## Generate a new Serial
+## Generate a new Serial and Board Serial (MLB) for your model.
+
+Compile [macserial](https://github.com/acidanthera/OpenCorePkg/tree/master/Utilities/macserial) from source.
+
+```bash
+git clone --depth 1 https://github.com/acidanthera/OpenCorePkg.git
+cd ./OpenCorePkg/Utilities/macserial/
+make
+chmod +x ./macserial
+```
+
+Find your **SystemProductName** in your config.plist file. That is your model number.
+
+Replace `"iMacPro1,1"` below with SystemProductName in your config.plist.
+
+```bash
+./macserial --num 1 --model "iMacPro1,1" 
+```
+Example output:
+
+```console
+$ ./macserial \
+        --model "iMacPro1,1" 
+Warning: arc4random is not available!
+C02V7UYGHX87 | C02733401J9JG36A8
+```
+
+The value on the left is your **Serial number**.
+The value on the right is your **Board Serial (MLB)**.
+
+## Choose a MAC Address 
+
+Select a MAC Address with an Organizationally Unique Identifier (OUI) that corresponds to a real Apple, Inc. interface.
+
+See the following list:
+
+[https://gitlab.com/wireshark/wireshark/-/raw/master/manuf](https://gitlab.com/wireshark/wireshark/-/raw/master/manuf)
+
+For example:
+
+```
+00:16:CB    Apple   Apple, Inc.
+```
+
+Make up the last 3 octets.
+
+For example:
+```
+00:16:CB:00:11:22
+```
+
+## Derive the corresponding ROM Value 
+
+ROM is calculated from your MAC Address.
+
+Lowercase your MAC Address, and remove each colon `:` between the octets.
+
+For example:
+
+**MAC:** `00:16:CB:00:11:22`
+
+**ROM:** `0016cb001122`
+
+## Generate an UUID
+
+Type `uuidgen` in Terminal
+
+```console
+$ uuidgen
+976AA603-75FC-456B-BC6D-9011BFB4968E
+```
+
+Then simply replace those values in your config.plist in the same section:
+
+```xml
+    <key>MLB</key>
+    <string>C02733401J9JG36A8</string>
+    <key>ROM</key>
+    <data>0016cb001122</data>
+    <key>SpoofVendor</key>
+    <true/>
+    <key>SystemProductName</key>
+    <string>iMacPro1,1</string>
+    <key>SystemSerialNumber</key>
+    <string>C02V7UYGHX87</string>
+    <key>SystemUUID</key>
+    <string>976AA603-75FC-456B-BC6D-9011BFB4968E</string>
+```
+
+NOTE: If you have trouble using the App Store, you [may need to fix En0](#fixing-en0), depending on your hardware setup.
```

Comment:
You forgot to mention what values from Apple's site the user wants, ie. Invalid, Valid, in use

Reply:
I honestly don't check them anymore because I believe it doesn't matter: all 3 types of support validity work.

Reply:
In my account right now, I have two invalid serials, and a third one which has a valid purchase date.

Both of the invalid serial numbers are running Big Sur, and using iServices with no issues.

Reply:
How the page is formatted now, doesn't require you to add an explicit serial check mention. So resolved